app/controllers/actions/SessionIdAction.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.actions
+
+import javax.inject.Inject
+import play.api.mvc.request.Cell
+import play.api.mvc.request.RequestAttrKey
+import play.api.mvc._
+import uk.gov.hmrc.http.SessionKeys
+import play.api.Logger
+import scala.concurrent.{ExecutionContext, Future}
+
+trait SessionIdAction extends ActionBuilder[Request, AnyContent] with ActionFunction[Request, Request] {
+  val EgNewSessionIdName: String = "EG_NEW_SESSIONID"
```

Comment:
I'd start with a lower case "e"; this isn't C# you know.

Reply:
Its Scala :) https://docs.scala-lang.org/style/naming-conventions.html

Reply:
They call it upper camel case i.e. camel case beginning with an initial upper case char